### ISSUE TITLE
Permissions update aren't being saved correctly

### ISSUE THREAD

**Issue by salsakran** [metabase/metabase#3888]
From @mazameli's comment in #3867:

Also, when trying to grant selective access to more than a single table in a DB for which a group has Limited Access, when saving the permissions changes, only the table(s) for which they have newly been granted access will show up as green; if they had access to any other table(s), that access gets revoked.

From @tlrobinson's comment in #3867:

Regarding the second half ("only the table(s) for which they have newly been granted access will show up as green; if they had access to any other table(s), that access gets revoked"), this appears to be a backend bug.

I start with the following permissions for a particular db/group:

```javascript
{native: "none", schemas: {PUBLIC: {1: "all", 2: "none", 3: "none", 4: "none"}}}
```

In the UI I grant permission to table 3, and hit save. The following is POSTed:

```javascript
{native: "none", schemas: {PUBLIC: {1: "all", 2: "none", 3: "all", 4: "none"}}}
```

And the following is persisted and returned:

```javascript
{native: "none", schemas: {PUBLIC: {1: "none", 2: "none", 3: "all", 4: "none"}}}
```

**Comment by dacort** [comment]
Probably the same as #3718, which didn't get picked up so could probably close that in favor of this if that's the case.

**Comment by camsaul** [comment]
Fixed by #3891
